cmd/kubeadm/app/phases/token/bootstrap.go

```diff
@@ -81,12 +85,55 @@ func UpdateOrCreateToken(client *clientset.Clientset, d *kubeadmapi.TokenDiscove
 	)
 }
 
+// CreateBootstrapConfigMap creates the public cluster-info ConfigMap
+func CreateBootstrapConfigMap(file string) error {
+	adminConfig, err := clientcmd.LoadFromFile(file)
+	if err != nil {
+		return fmt.Errorf("failed to load admin kubeconfig [%v]", err)
+	}
+	client, err := kubeconfigutil.KubeConfigToClientSet(adminConfig)
+	if err != nil {
+		return err
+	}
+
+	adminCluster := adminConfig.Contexts[adminConfig.CurrentContext].Cluster
+	// Copy the cluster from admin.conf to the bootstrap kubeconfig, contains the CA cert and the server URL
+	bootstrapConfig := &clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			adminCluster: adminConfig.Clusters[adminCluster],
```

Comment:
The spec says "It is expected that the `kubeconfig` file will have a single unnamed `Cluster` entry.  Other information (especially authentication secrets) MUST be omitted."

This translates to an empty key here I think.